src/js/profile/mobile/widget/TimePicker.js

```diff
@@ -261,8 +347,175 @@
 				time.setMinutes(parseInt(spins.minute.value(), 10));
 
 				return time;
+			};
+
+			/**
+			* Handle the click event
+			* @method _onClick
+			* @param {Event} event
+			* @member ns.widget.mobile.TimePicker
+			* @protected
+			*/
+			prototype._onClick = function (event) {
+				var self = this,
+					ui = self._ui,
+					element = self.element,
+					eventTargetElement = event.target,
+					hourContainer = ui.hourContainer,
+					minuteContainer = ui.minuteContainer,
+					parentContainer = getClosestByClass(eventTargetElement, TimePicker.classes.CONTAINER);
+
+				if (parentContainer && parentContainer.classList.contains(TimePicker.classes.HOUR_CONTAINER) &&
+					getClosestByClass(eventTargetElement, Spin.classes.SELECTED)) {
+					self._focusInput("hour");
+				} else if (parentContainer && parentContainer.classList.contains(TimePicker.classes.MINUTE_CONTAINER) &&
+					getClosestByClass(eventTargetElement, Spin.classes.SELECTED)) {
+					self._focusInput("minute");
+				} else {
+					element.classList.remove(TimePicker.classes.TIME_INPUT_ACTIVE);
+					minuteContainer.classList.remove(TimePicker.classes.ACTIVE_CONTAINER);
+					hourContainer.classList.remove(TimePicker.classes.ACTIVE_CONTAINER);
+				}
+			};
+
+			// Get a new added value
+			function diffValue(origin, change) {
+				var diff = change.toString().split(origin).join("");
+
+				if (diff.length === 0) {
+					return origin;
+				}
+				return diff;
 			}
 
+			/**
+			* Change the value of input element included in Spin
+			* @method _onInputChange
+			* @param {Event} event
+			* @member ns.widget.mobile.TimePicker
+			* @protected
+			*/
+			prototype._onInputChange = function (event) {
+				var self = this,
+					ui = self._ui,
+					options = self.options,
+					value = event.target.value,
+					activeContainer = getClosestByClass(event.target, TimePicker.classes.CONTAINER),
+					visibleValue;
+
+				if (activeContainer && activeContainer.classList.contains(TimePicker.classes.HOUR_CONTAINER)) {
+					if (!self._inputValueState) {
+						visibleValue = parseInt(diffValue(self._previousInputValue, value), 10);
+						self._setInputValue("hour", visibleValue, true);
+
```

Comment:
If you input 14 hours by SIP in the 12h format,  do you show the toast error message??

Reply:
+418							// TODO: Show toast error message
I will add the toast popup when finishing the implementaion of the toast.